apps/src/lib/node/ledger/mod.rs

```diff
@@ -748,3 +754,31 @@ async fn maybe_start_ethereum_oracle(
 fn spawn_dummy_task<T: Send + 'static>(ready: T) -> task::JoinHandle<T> {
     tokio::spawn(async { std::future::ready(ready).await })
 }
+
+/// Sleep until the genesis time if necessary.
+async fn sleep_until(time: DateTimeUtc) {
+    let shutdown_signal = install_shutdown_signal();
```

Comment:
this overrides another shutdown signal no? not sure if ctrl-c won't do anything afterwards

Reply:
When I tested locally, this worked fine. And reading the code, creates another shutdown handler after this one that the rest of the process use. However, to shutdown, I think you have to hit ctrl-c twice right now. I'll fix that.

Reply:
hm this still seems incorrect, because every time we call `install_shutdown_signal`, we spawn a new async task listening for interrupt signals. would it be better to expose the `ShutdownSignal` handler from the `AbortableSpawner`, and poll it from `tokio::select!()` instead?

Reply:
What's the problem with spawning an async task? No other shutdowns are install at the point and the task gets dropped at the completion of this call.

Reply:
`install_shutdown_signal` internally calls `tokio::spawn`, to listen to interrupt signals from the OS. by the end of `sleep_until`, if we return with `ControlFlow::Continue`, the process will have two async tasks running in the background listening for OS signals. I think it's unnecessary, bc/ we can re-use the handler inside of the `AbortableSpawner` to fetch incoming OS signals

Reply:
thx!